File: app/src/main/java/com/datarockets/mnchkn/activities/result/GameResultPresenterImpl.java
package com.datarockets.mnchkn.activities.result;

public class GameResultPresenterImpl implements GameResultPresenter,
        GameResultInteractorImpl.OnResultsLoaded {

    private GameResultView mGameResultView;
    private GameResultInteractor mGameResultInteractor;

    public GameResultPresenterImpl(GameResultView gameResultView,
                                   GameResultInteractor gameResultInteractor) {
        this.mGameResultView = gameResultView;
        this.mGameResultInteractor = gameResultInteractor;
    }


    @Override
    public void onCreate() {
        if (mGameResultView != null) {
            mGameResultInteractor.loadGameResults(this);
        }
    }

    @Override
    public void notifyChartDataPrepared() {
        if (mGameResultView != null) {
            mGameResultView.loadChartFragments();
        }
    }

    @Override
    public void onBackPressed() {
        if (mGameResultView != null) {
            mGameResultInteractor.clearSteps();
            mGameResultInteractor.clearPlayerStats();
        }
    }

    @Override
    public void onStop() {
        if (mGameResultView != null) {
            mGameResultInteractor.clearSteps();
            mGameResultInteractor.clearPlayerStats();
        }
    }

    @Override
    public void onDestroy() {
        if (mGameResultView != null) {
            mGameResultInteractor.clearSteps();
            mGameResultInteractor.clearPlayerStats();
            mGameResultView = null;
        }
    }

}

File: app/src/main/kotlin/com/datarockets/mnchkn/activities/result/GameResultPresenterImpl.kt
package com.datarockets.mnchkn.activities.result

class GameResultPresenterImpl(private var mGameResultView: GameResultView?,
                              private val mGameResultInteractor: GameResultInteractor) : GameResultPresenter, GameResultInteractor.OnResultsLoaded {


    override fun onCreate() {
        mGameResultInteractor.loadGameResults(this)
    }

    override fun notifyChartDataPrepared() {
        mGameResultView?.loadChartFragments()
    }

    override fun onBackPressed() {
        mGameResultInteractor.clearSteps()
        mGameResultInteractor.clearPlayerStats()
    }

    override fun onStop() {
        mGameResultInteractor.clearSteps()
        mGameResultInteractor.clearPlayerStats()
    }

    override fun onDestroy() {
        if (mGameResultView != null) {
            mGameResultInteractor.clearSteps()
            mGameResultInteractor.clearPlayerStats()
            mGameResultView = null
        }
    }

}
